core/settings/qPrefUnit.cpp

```diff
@@ -28,15 +28,63 @@ void qPrefUnits::loadSync(bool doSync)
 
 HANDLE_PREFERENCE_BOOL(Units, "coordinates", coordinates_traditional);
 
-HANDLE_PREFERENCE_ENUM_EXT(Units, units::DURATION, "duration_units", duration_units, units.);
+QString qPrefUnits::duration_units()
+{
+	return 	prefs.units.duration_units == units::DURATION::ALWAYS_HOURS ? QStringLiteral("hours") :
+			prefs.units.duration_units == units::DURATION::MINUTES_ONLY ? QStringLiteral("minutes") :
+											QStringLiteral("mixed");
+}
+void qPrefUnits::set_duration_units(const QString& value)
+{
+	prefs.units.duration_units = value == QStringLiteral("hours") ? units::DURATION::ALWAYS_HOURS :
+								 value == QStringLiteral("minutes")? units::DURATION::MINUTES_ONLY :
+																	 units::DURATION::MIXED;
+	disk_duration_units(true);
+	emit instance()->duration_unitsChanged(value);
+}
+SET_PREFERENCE_ENUM_EXT(Units, units::DURATION, duration_units, units.);
+DISK_LOADSYNC_ENUM_EXT(Units, "duration_units", units::DURATION, duration_units, units.);
```

Comment:
so we had something that works (at least I'm not aware of it not working). And we replace it with something that seems a lot more verbose and confusing. Why?

Reply:
to have units as strings in QML instead of as int (that are really an enum).

Reply:
Maybe the commit message could explain this a little more clearly.
"Instead of passing arounds enum/int value, it seems easier to pass string literals to QML and have that code respond to those"
And then have the corresponding QML change in the SAME commit.

Reply:
updated commit message. The QML change is not in this PR, it is part of the dive planner PRs